File: codeforces/71A/A_Way_Too_Long_Words.java
import java.util.* ;


class A_Way_Too_Long_Words{
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int t = s.nextInt();
        for (int i = 0; i <= t; i++){
            String word1 = s.nextLine();
            solve(word1);
        }
        s.close();
    }

    static void solve(String word) {
        int n = word.length();
        if (n > 10) {
            char begin = word.charAt(0);
            char end = word.charAt(n-1);
            Integer a = n-2;
            System.out.println(begin + a.toString() + end);
            return;
        } else {
            System.out.println(word);
            return;
        }
    }
}

File: codeforces/71A/A_Way_Too_Long_Words.cpp
#include<bits/stdc++.h>
using namespace std;

void solve(string word) {
    int n = word.length();
    if (n > 10) {
        char begin = word.at(0);
        char end = word.at(n-1);
        cout << begin << n-2 << end << endl;
    } else {
        cout << word << endl;
    }

}
int main() {
    int t;
    cin >> t;
    for (int i = 0; i < t; i++) {
        string word;
        cin >> word;
        solve(word);
    }

    return 0;
}